frontend/src/views/assignments/import-export.js

```diff
@@ -86,6 +86,112 @@ function formatWageChunksToList(wageChunks) {
     return ret;
 }
 
+/**
+ * Returns a function which converts a list of selected assignments into a File object
+ *
+ * @export
+ * @returns
+ */
+export function prepareData(
+    selectedAssignmentIds,
+    session,
+    createPayPeriodHeaders,
+    formatWageChunksToList,
+    formatDateForSpreadsheet
+) {
```

Comment:
Don't do this...

Make a `prepareDataFactory(filterFunc)` function which returns a `prepareData` function that looks like all the others, except, `filterFunc` is applied to the data before the rest of the processing.

`filterFunc` should be optional and default to the identity function.

Reply:
created prepareDataFactory and refactored the object's prepareData feature pretty much.